pkg/iac/terraform/state/backend/http_reader_test.go

```diff
@@ -28,52 +32,91 @@ func TestNewHTTPReader(t *testing.T) {
 					Headers: map[string]string{},
 				},
 			},
-			wantURL: "",
 			wantErr: errors.New("Get \"wrong_url\": unsupported protocol scheme \"\""),
+			httpClient: func() HttpClient {
+				return &http.Client{}
+			}(),
+			expected: "",
 		},
 		{
 			name: "Should fetch URL with auth header",
 			args: args{
-				url: "https://raw.githubusercontent.com/cloudskiff/driftctl/main/.dockerignore",
+				url: "https://wrong.url/cloudskiff/driftctl/main/terraform.tfstate",
 				options: &Options{
 					Headers: map[string]string{
 						"Authorization": "Basic Test",
 					},
 				},
 			},
-			wantURL: "https://raw.githubusercontent.com/cloudskiff/driftctl/main/.dockerignore",
 			wantErr: nil,
+			httpClient: func() HttpClient {
+				m := &mocks.HttpClient{}
+
+				req, _ := http.NewRequest(http.MethodGet, "https://wrong.url/cloudskiff/driftctl/main/terraform.tfstate", nil)
+
+				req.Header.Add("Authorization", "Basic Test")
+
+				bodyReader := strings.NewReader("{}")
+				bodyReadCloser := io.NopCloser(bodyReader)
+
+				m.On("Do", req).Return(&http.Response{
+					StatusCode: 200,
+					Body:       bodyReadCloser,
+				}, nil)
+
+				return m
+			}(),
+			expected: "{}",
 		},
 		{
 			name: "Should fail with bad status code",
 			args: args{
-				url: "https://raw.githubusercontent.com/cloudskiff/driftctl-test/main/.dockerignore",
+				url: "https://wrong.url/cloudskiff/driftctl/main/terraform.tfstate",
```

Comment:
This is very minor, but you can use `example.com` for this purpose, it is what it has been [designed for](https://www.iana.org/domains/reserved).